### PR TITLE
Restored partial backwards compatibility with 2.x

### PR DESCRIPTION
**Pull request by agronholm**
The list of truly breaking changes has been narrowed down quite a bit.

### CODE REVIEW

**Comment by smurfix** [comment]
Nice. Saves me some work. *g*

**Comment by agronholm** [comment]
Can you do a review/approval here?

File: docs/versionhistory.rst [hunk 1]
```diff
@@ -12,26 +12,30 @@ This library adheres to `Semantic Versioning 2.0 <http://semver.org/>`_.
 
   * ``current_time()``
   * ``current_effective_deadline()``
+  * ``get_current_task()``
```
Comment (smurfix) [inline docs/versionhistory.rst:15]: For even more improved fake-await-ery we could add ```
def __await__(self):
    if False: yield
    warn("Don't do this")
    return self
```
method to `ansio.TaskInfo`. That'd fix `get_current_task`. A list with that method tacked on is almost as simple, for `get_running_tasks`.

Comment (agronholm) [reply docs/versionhistory.rst:15]: Ok, that's fair.

**Comment by agronholm** [comment]
As a bonus, `get_running_tasks()` is also now backwards compatible.

**Review by smurfix — APPROVED** [review]

File: src/anyio/_core/_compat.py [hunk 2]
```diff
@@ -76,6 +76,16 @@ def __await__(self):
             yield
 
 
+class DeprecatedAwaitableList(Generic[T], List[T], DeprecatedAwaitable):
```
Comment (smurfix) [inline src/anyio/_core/_compat.py:79]: Huh. Do we actually need the `Generic[T]` part here?

Comment (agronholm) [reply src/anyio/_core/_compat.py:79]: Yes, it's needed for any class that requires a type argument.

Comment (graingert) [reply src/anyio/_core/_compat.py:79]: I thought if you already inherited from a GenericAlias it worked:

https://mypy-play.net/?mypy=latest&python=3.9&gist=fdc3ca168923e66527952336ccf7f320

Comment (agronholm) [reply src/anyio/_core/_compat.py:79]: I couldn've sworn that I tried this already. I guess I left out the type argument.

**Pull request merged**